auto_tests/toxav_mt_test.cc

```diff
@@ -0,0 +1,498 @@
+#include <algorithm>
+#include <atomic>
+#include <chrono>
+#include <cstdio>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+#include "../testing/misc_tools.h"
+#include "../toxav/toxav.h"
+#include "../toxcore/tox.h"
+#include "check_compat.h"
+
+// Make lines shorter
+using Clock = std::chrono::high_resolution_clock;
+using TimePoint = std::chrono::time_point<Clock>;
+
+namespace {
+// Maximum amount of time in iterations to wait for bootstrapping and friend
+// connections to succeed.
+constexpr uint32_t MAX_BOOTSTRAP_ITERATIONS = 1000;
+
+struct Tox_Deleter {
+  void operator()(Tox *tox) const { tox_kill(tox); }
+};
+
+using Tox_Ptr = std::unique_ptr<Tox, Tox_Deleter>;
+
+struct ToxAV_Deleter {
+  void operator()(ToxAV *tox) const { toxav_kill(tox); }
+};
+
+using ToxAV_Ptr = std::unique_ptr<ToxAV, ToxAV_Deleter>;
+
+static void t_accept_friend_request_cb(Tox *m, const uint8_t *public_key, const uint8_t *data,
+                                       size_t length, void *userdata) {
+  if (length == 7 && std::memcmp("gentoo", data, 7) == 0) {
```

Comment:
Do we expect other friend requests to arrive? Maybe we should assert that we don't.

Reply:
Should be fixed now